packages/cli/src/deploy/ensureModules.ts

```diff
@@ -17,49 +17,47 @@ export async function ensureModules({
   if (!modules.length) return [];
 
   debug("installing modules:", modules.map((mod) => mod.name).join(", "));
-  const installTxs = await Promise.all(
-    modules.map((mod) =>
-      mod.installAsRoot
-        ? pRetry(
-            () =>
-              writeContract(client, {
-                chain: client.chain ?? null,
-                address: worldDeploy.address,
-                abi: worldAbi,
-                // TODO: replace with batchCall (https://github.com/latticexyz/mud/issues/1645)
-                functionName: "installRootModule",
-                args: [mod.address, mod.installData],
-              }),
-            {
-              retries: 3,
-              onFailedAttempt: async (error) => {
-                const delay = error.attemptNumber * 500;
-                debug(`failed to install root module ${mod.name}, retrying in ${delay}ms...`);
-                await wait(delay);
-              },
+  return (
+    await Promise.all(
+      modules.map((mod) =>
+        pRetry(
+          async () => {
+            try {
+              return mod.installAsRoot
+                ? await writeContract(client, {
+                    chain: client.chain ?? null,
+                    address: worldDeploy.address,
+                    abi: worldAbi,
+                    // TODO: replace with batchCall (https://github.com/latticexyz/mud/issues/1645)
+                    functionName: "installRootModule",
+                    args: [mod.address, mod.installData],
+                  })
+                : await writeContract(client, {
+                    chain: client.chain ?? null,
+                    address: worldDeploy.address,
+                    abi: worldAbi,
+                    // TODO: replace with batchCall (https://github.com/latticexyz/mud/issues/1645)
+                    functionName: "installModule",
+                    args: [mod.address, mod.installData],
+                  });
+            } catch (error) {
+              if (error instanceof BaseError && error.message.includes("Module_AlreadyInstalled")) {
```

Comment:
I wanted to put this in `onFailedAttempt` but there's no way to signal to `p-retry` that we should stop retrying and continue.